File: src/test/java/com/morak/performancetracker/aop/persistence/StatementAdvisorTest.java
package com.morak.performancetracker.aop.persistence;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.*;

import com.morak.performancetracker.context.Accumulator;
import com.morak.performancetracker.context.ResultMapper;
import org.junit.jupiter.api.Test;

class StatementAdvisorTest {

    @Test
    void statement_advisor를_생성한다() {
        // given
        StatementPointcut pointcut = new StatementPointcut();
        StatementAdvice advice = new StatementAdvice(new QueryMonitor(), new Accumulator(new ResultMapper()));

        // when
        StatementAdvisor advisor = new StatementAdvisor(pointcut, advice);

        // then
        assertThat(advisor).isNotNull();
    }

}

File: src/test/kotlin/com/morak/performancetracker/aop/persistence/StatementAdvisorTest.kt
package com.morak.performancetracker.aop.persistence

import com.morak.performancetracker.context.Accumulator
import com.morak.performancetracker.context.ResultMapper
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class StatementAdvisorTest {
    @Test
    fun `statement advisor를 생성한다`() {
        // given
        val pointcut = StatementPointcut()
        val advice = StatementAdvice(QueryMonitor(), Accumulator(ResultMapper()))

        // when
        val advisor = StatementAdvisor(pointcut, advice)

        // then
        Assertions.assertThat(advisor).isNotNull()
    }
}
